File: projects/poly/code/halve.go
package main

import "fmt"

// halve the input, rounded down,
// never reaching 0, e.g.:
//   5
// returns:
//   2
func halve(x int) int {
	v := x / 2
	if v < 1 {
		v = 1
	}

	return v
}

func main() {
	fmt.Printf("%d\n", halve(5))
}

File: projects/poly/code/halve.js
// halve the input, rounded down,
// never reaching 0, e.g.:
//   5
// returns:
//   2
function halve(x) {
  let v = parseInt(x / 2);
  if (v < 1) {
    v = 1
  }

  return v;
}

console.log(JSON.stringify(halve(5)));
